### PR TITLE
[PT FE] Make all pytorch layer tests use static seed for random numbers

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Implement test RNG via `SeededRandom`, defaulting to FP32 outputs and consistent seed usage to stabilize test inputs.*
- *Update layer tests to use RNG helpers and dtype parameters directly instead of ad‑hoc casts.*
- *Remove unused imports and minor cleanup across PyTorch tests*
- *Adjust string constant handling in `fx_decoder.py`.*

### Tickets:
 - 181579

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_grid_sampler.py [hunk 1]
```diff
@@ -10,31 +10,29 @@
...
-        ref_net = None
 
-        return aten_grid_sampler(mode, padding_mode, align_corners), ref_net, "aten::grid_sampler"
+        return aten_grid_sampler(mode, padding_mode, align_corners), "aten::grid_sampler"
 
     @pytest.mark.parametrize(["h_in", "w_in", "h_out", "w_out"], ([10, 10, 5, 5], [10, 15, 3, 5]))
     @pytest.mark.parametrize("mode", ["bilinear", "nearest", "bicubic"])
     @pytest.mark.parametrize("padding_mode", ["zeros", "border", "reflection"])
-    @pytest.mark.parametrize("align_corners", [True, False, None])
+    @pytest.mark.parametrize("align_corners", [True, False])
```
[the remark this answers is not in the document]
Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_grid_sampler.py:35]: None is deprecated, it was deleted to remove warning

File: tests/layer_tests/pytorch_tests/test_strided_const.py [hunk 2]
```diff
@@ -9,22 +9,22 @@
...
     def create_model(self):
         import torch
 
         class strided_const(torch.nn.Module):
-            def __init__(self):
+            def __init__(self, rng):
                 super().__init__()
-                self.const = torch.randint(0, 10, [1, 3, 2, 2], dtype=torch.float32)
+                self.const = rng.torch_randint(0, 10, [1, 3, 2, 2], dtype=torch.float32)
                 self.const = self.const.to(memory_format=torch.channels_last)
```
[the remark this answers is not in the document]
Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_strided_const.py:22]: it uses .to for type conversion

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 3]
```diff
@@ -19,6 +19,121 @@
...
+        """Initialize with a seed value."""
+        self._seed = seed
+        self._rng = np.random.RandomState(seed)
+
+    @staticmethod
+    def _to_numpy_dtype(dtype):
+        if dtype is None:
+            return None
+        if isinstance(dtype, torch.dtype):
+            dtype_map = {
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:40]: [nitpick] This is static data and can sit in the module scope:
```
# pytorch_layer_test_class.py‎
TORCH_TO_NUMPY_TYPE = {
    torch.float32: np.float32,
    torch.float64: np.float64,
    torch.float16: np.float16,
    torch.int8: np.int8,
    torch.int16: np.int16,
    torch.int32: np.int32,
    torch.int64: np.int64,
    torch.uint8: np.uint8,
    torch.bool: np.bool_,
}

class SeededRandom:
    ...

    @staticmethod
    def _to_numpy_dtype(dtype):
        if dtype is None:
            return None
        if isinstance(dtype, torch.dtype):
           return TORCH_TO_NUMPY_TYPE.get(dtype, np.float32)
```

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 4]
```diff
@@ -19,6 +19,121 @@
...
+                torch.float64: np.float64,
+                torch.float16: np.float16,
+                torch.int8: np.int8,
+                torch.int16: np.int16,
+                torch.int32: np.int32,
+                torch.int64: np.int64,
+                torch.uint8: np.uint8,
+                torch.bool: np.bool_,
+            }
+            return dtype_map.get(dtype, np.float32)
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:51]: I guess if we can't perform a mapping, this indicates some broken invariant (e.g. missing  code for new types or code accidentally broken in maintenance).

I guess this should be assertive, i.e. throw if we're trying to map an unknown type.
In the simplest form:
```
return dtype_map[dtype]
```
which, unfortunately, throws a cryptic (too low level) `KeyError` exception. So if you find this inelegant, raise a more meaningful exception.

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 5]
```diff
@@ -19,6 +19,121 @@
...
+    tests to be run in any order or in parallel while maintaining reproducibility.
+    """
+    
+    def __init__(self, seed: int = 0):
+        """Initialize with a seed value."""
+        self._seed = seed
+        self._rng = np.random.RandomState(seed)
+
+    @staticmethod
+    def _to_numpy_dtype(dtype):
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:36]: [nitpick] The Hungarian notation here doesn't really improve access rights (everything is public in Python anyway) and, in my humble opinion, only adds syntactic noise. 
How about simply:
```suggestion
    def to_numpy_dtype(dtype):
```
?

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 6]
```diff
@@ -19,6 +19,121 @@
...
+                torch.int64: np.int64,
+                torch.uint8: np.uint8,
+                torch.bool: np.bool_,
+            }
+            return dtype_map.get(dtype, np.float32)
+        return dtype
+    
+    def randn(self, *shape, dtype=None) -> np.ndarray:
+        """Generate random samples from standard normal distribution."""
+        np_dtype = self._to_numpy_dtype(dtype) if dtype is not None else np.float32
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:56]: How about having the whole type calculation done in `_to_numpy_dtype()`:
```
@staticmethod
def _to_numpy_dtype(dtype, dtype_default):
    # Just return `dtype_default` in places you're returning `None` at the moment 
```

This should simplify your logic a little bit.

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 7]
```diff
@@ -19,6 +19,121 @@
...
+        return np.asarray(self._rng.randn(*shape)).astype(np_dtype)
+    
+    def rand(self, *shape, dtype=None) -> np.ndarray:
+        """Generate random samples from uniform distribution [0, 1)."""
+        np_dtype = self._to_numpy_dtype(dtype) if dtype is not None else np.float32
+        return np.asarray(self._rng.rand(*shape)).astype(np_dtype)
+    
+    def randint(self, low, high=None, size=None, dtype=np.int64):
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:64]: Question: are signatures of these methods taken from `numpy` itself?
As we have some inconsistency potentially leading to surprises, e.g:
```
# Calling seed_random.rand(..., dtype=None) plays nicely and returns an inferred type
def rand(self, *shape, dtype=None) -> np.ndarray:
    ...

# Calling seed_random.randint(..., dtype=None) bypasses type normalisation
def randint(self, low, high=None, size=None, dtype=np.int64):
    ...
```

If not influenced by `numpy` itself, consider unifying the behaviour (preferably by having `None` as a default `dtype`).
